File: source/programing_exercises/java/src/test/java/FizzBuzzTest.java
import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class FizzBuzzTest {
    private FizzBuzz _fizzBuzz;
    private FizzBuzzType _typeNormal;
    private FizzBuzzType _typeOne;
    private FizzBuzzType _typeTwo;
    private FizzBuzzType _typeThree;
    private FizzBuzzType _typeFour;
    private FizzBuzzType _typeFive;
    
    @BeforeEach
    void init() {
        _fizzBuzz = new FizzBuzz();
        _typeNormal = FizzBuzzType.valueOf("normal");
        _typeOne = FizzBuzzType.valueOf("one");
        _typeTwo = FizzBuzzType.valueOf("two");
        _typeThree = FizzBuzzType.valueOf("three");
        _typeFour = FizzBuzzType.valueOf("four");
        _typeFive = FizzBuzzType.valueOf("five");    
    }    
    @Test
    public void 数が3ならばFizzを返す() {        
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeNormal);
        command.execute(3);
        assertEquals("Fizz", command.getValue());
    }
    @Test
    public void 数が5ならばBuzzをかえす() {
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeNormal);
        command.execute(5);        
        assertEquals("Buzz", command.getValue());
    }    
    @Test
    public void 数が15ならばFizzBuzzを返す() {  
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeNormal);
        command.execute(15);        
        assertEquals("FizzBuzz", command.getValue());
    }
    @Test
    public void 数が1ならば1を返す() {  
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeNormal);
        command.execute(1);                      
        assertEquals("1", command.getValue());
    }
    @Test
    public void 数が101ならば101を返す() {      
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeNormal);
        command.execute(101);          
        assertEquals("101", command.getValue());
    }
    @Test
    public void 回数を5回繰り返し実行したならば配列を返す() {
        String[] expected = {"1","2","Fizz","4","Buzz"};
        assertArrayEquals(expected, _fizzBuzz.iterate(5));
    }
    @Test
    public void 回数を10回繰り返し実行したならば配列を返す() {
        String[] expected = {"1","2","Fizz","4","Buzz", "Fizz", "7", "8", "Fizz", "Buzz"};
        assertArrayEquals(expected, _fizzBuzz.iterate(10));
    }    
    @Test
    public void タイプ１は数を返す() {    
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeOne);
        command.execute(3);                                     
        assertEquals("3", command.getValue());
    }
    @Test
    public void タイプ２はFizzを返す() {       
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeTwo);
        command.execute(1);                                               
        assertEquals("Fizz", command.getValue());
    }
    @Test
    public void タイプ３はBuzzを返す() {   
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeThree);
        command.execute(1);                                               
        assertEquals("Buzz", command.getValue());
    }    
    @Test
    public void タイプ４は大文字の値を返す() { 
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeFour);
        command.execute(3);                                               
        assertEquals("FIZZ", command.getValue());        
        command.execute(5);                                               
        assertEquals("BUZZ", command.getValue());
        command.execute(15);                                               
        assertEquals("FIZZBUZZ", command.getValue());
    } 
    @Test
    public void タイプ５は２で割り切れたらFizz３で割り切れたらBuzz２と３で割り切れたならFIZZBUZZを返す() {                
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeFive);
        command.execute(2);
        assertEquals("Fizz", command.getValue());        
        command.execute(3);
        assertEquals("Buzz", command.getValue());             
        command.execute(6);        
        assertEquals("FIZZBUZZ", command.getValue());        
    } 
    @Test
    public void 上記以外のは通常のFizzBuzzパターンを返す() {  
        FizzBuzzValueCommand command = new FizzBuzzValueCommand(_typeNormal);        
        command.execute(3);
        assertEquals("Fizz", command.getValue());    
        command.execute(5);
        assertEquals("Buzz", command.getValue());                
        command.execute(15);
        assertEquals("FizzBuzz", command.getValue());        
    }    
}

File: source/programing_exercises/c_sharp/FizzBuzzService.Tests/FizzBuzzTest.cs
using System;
using System.Collections.Generic;
using Xunit;

namespace FizzBuzzService.Tests
{
    public class FizzBuzzTest : IDisposable
    {
        FizzBuzz _fizzBuzz;
        FizzBuzzType _typeOne;
        FizzBuzzType _typeTwo;
        FizzBuzzType _typeThree;
        FizzBuzzType _typeFour;
        FizzBuzzType _typeFive;
        FizzBuzzType _typeStandard;
        public FizzBuzzTest()
        {     
            _fizzBuzz = new FizzBuzz();
            _typeOne = FizzBuzzType.One;
            _typeTwo = FizzBuzzType.Two;
            _typeThree = FizzBuzzType.Three;
            _typeFour = FizzBuzzType.Four;
            _typeFive = FizzBuzzType.Five;
            _typeStandard = FizzBuzzType.Standard;
        }
        public void Dispose(){}        

        [Fact]
        public void 値が３ならばFizzを返す()
        {                             
            FizzBuzz fizzBuzz = new FizzBuzz(_typeStandard);
            FizzBuzzValue actual = fizzBuzz.generate(3);            
            Assert.Equal("Fizz", actual.Value);            
            Assert.Equal(3, actual.Number);
        }
        [Fact]
        public void 値が５ならばBuzzを返す()
        {
            FizzBuzz fizzBuzz = new FizzBuzz(_typeStandard);
            FizzBuzzValue actual = fizzBuzz.generate(5);            
            Assert.Equal("Buzz", actual.Value);
            Assert.Equal(5, actual.Number);
        }
        [Fact]
        public void 値が１５ならばFizzBuzzを返す()
        {
            FizzBuzz fizzBuzz = new FizzBuzz(_typeStandard);
            FizzBuzzValue actual = fizzBuzz.generate(15);            
            Assert.Equal("FizzBuzz", actual.Value);
            Assert.Equal(15, actual.Number);
        }       
        [Fact]         
        public void 値が１ならば１を返す()
        {
            FizzBuzz fizzBuzz = new FizzBuzz(_typeStandard);
            FizzBuzzValue actual = fizzBuzz.generate(1);            
            Assert.Equal("1", actual.Value);
            Assert.Equal(1, actual.Number);
        }
        [Fact]
        public void 値が１０１ならば１０１を返す()
        {
            FizzBuzz fizzBuzz = new FizzBuzz(_typeStandard);
            FizzBuzzValue actual = fizzBuzz.generate(101);                        
            Assert.Equal("101", actual.Value);
            Assert.Equal(101, actual.Number);            
        }        
        [Fact]
        public void 回数を５回繰り返し実行したら配列を返す()
        {
            _fizzBuzz.iterate(5);
            string[] expect = {"1","2","Fizz","4","Buzz"};
            Assert.Equal(expect, _fizzBuzz.Values.arrayValue());
        }
        [Fact]
        public void 回数を１０回繰り返し実行したら配列を返す()
        {
            _fizzBuzz.iterate(10);
            string[] expect = {"1","2","Fizz","4","Buzz","Fizz","7","8","Fizz","Buzz"};
            Assert.Equal(expect, _fizzBuzz.Values.arrayValue());
        }
        [Fact]
        public void タイプ１は値を返す()
        {
            FizzBuzzValue actual = _typeOne.generate(3);
            Assert.Equal("3", actual.Value);
        }
        [Fact]
        public void タイプ２はFizzだけを返す()
        {
            FizzBuzzValue actual = _typeTwo.generate(5);
            Assert.Equal("Fizz", actual.Value);
        }     
        [Fact]   
        public void タイプ３はBuzzだけを返す()
        {            
            FizzBuzzValue actual = _typeThree.generate(3);
            Assert.Equal("Buzz", actual.Value);
        }   
        [Fact]
        public void タイプ４は通常パターンを大文字に変換して返す()
        {
            FizzBuzzValue actual = _typeFour.generate(3);
            Assert.Equal("FIZZ", actual.Value);
            actual = _typeFour.generate(5);
            Assert.Equal("BUZZ", actual.Value);            
            actual = _typeFour.generate(15);
            Assert.Equal("FIZZBUZZ", actual.Value);            
        }   
        [Fact]
        public void タイプ５は２で割り切れたらFizz３で割り切れたらBuzz２と３で割り切れたならFIZZBUZZを返す()
        {
            FizzBuzzValue actual = _typeFive.generate(2);
            Assert.Equal("Fizz", actual.Value);
            actual = _typeFive.generate(3);            
            Assert.Equal("Buzz", actual.Value);                        
            actual = _typeFive.generate(6);            
            Assert.Equal("FIZZBUZZ", actual.Value);            
        }                  
    }
}
